foursight_core/react/api/react_ui.py

```diff
@@ -0,0 +1,102 @@
+import os
+import io
+
+REACT_BASE_DIR = os.path.join(os.path.dirname(__file__), "../../react/ui")
+REACT_DEFAULT_FILE = "index.html"
+REACT_WHITELISTED_FILE_PATH_SUFFIXES = [
+    "/index.html",
+    "/main.js",
+    "/main.css",
+    "/manifest.json",
+    "/asset-manifest.json",
+    ".jpeg",
+    ".jpg",
+    ".png",
+    ".ico"
+]
+
+
+class ReactUi:
+
+    def __init__(self, react_api):
+        self.react_api = react_api
+
+    cache_static_files = {}
+
+    def serve_static_file(self, env: str, **kwargs):
+
+        # TODO: commentary on this.
+        env = env.replace("{environ}", env)
+
+        if env == "static":
+            # If the env is 'static' then we take this to mean the 'static'
+            # subdirectory; this is the directory where the static (js, css, etc)
+            # React files live. Note that this means a real 'environ' may not be 'static'.
+            file = os.path.join(REACT_BASE_DIR, "static")
+        else:
+            file = REACT_BASE_DIR
+        args = kwargs.values()
+        if not args:
+            # TODO: png (et.al.) not downloading right!
+            # Running chalice local it works though.
+            # Actually it also works in cgap-supertest:
+            # https://810xasmho0.execute-api.us-east-1.amazonaws.com/api/react/logo192.png
+            # But not in 4dn/foursight-development:
+            # https://cm3dqx36s7.execute-api.us-east-1.amazonaws.com/api/react/logo192.png
+            # Anyways for now the React UI references images at external sites not from here.
+            if (env.endswith(".js") or env.endswith(".html") or env.endswith(".json")
+               or env.endswith(".jpg") or env.endswith(".jpeg")
+               or env.endswith(".png") or env.endswith(".ico")):
```

Comment:
You could use `any` here to good effect. And you could factor out the list of suffixes into a constant, which actually might want to be a dictionary with these as keys. See my comments below.
```suggestion
            # Anyways for now the React UI references images at external sites not from here.
            if any(env.endswith(suffix) for suffix in [".js", ".html", ".json", ".jpg", ".jpeg", ".png", ".ico"]:
```

Reply:
Ah, nice, `any` is a new one on me, thanks.